### ISSUE TITLE
Adding editable layer to a new project

### ISSUE THREAD

**Issue by jshirokov-bs** [qwc-services/qwc-docker#59]
Hello.

I can add a new project to QWC2 using WFS. But how can I add an editable layer to a new project? If I add an editable layer from a demo project to my own project from Postgres, an error occurs during configuration.

**Comment by manisandro** [comment]
For editing, you need a dataset stored in a postgis DB which is added to your qgis project as a layer. Additionally, you need to set write permissions to the respective data resources in the admin gui - look at the demo edit_points/edit_lines/edit_polygons for an example.

**Comment by jshirokov-bs** [comment]
@tpo 

As a test example, I simply add the "edit_points" layer from the demo project (directly from postgre) to the new "world13" project , place the .qgs in the "scan" folder and start configuring services. After that, this error occurs.

![image](https://user-images.githubusercontent.com/117968115/228246209-be026356-6175-48ee-baca-29712c715f82.png)

**Comment by manisandro** [comment]
No, you need to use a service definition (as defined in pg_service.conf), as the necessary connection information differs inside the docker from the host system (i.e. in the container the DB can be found at `qwc-postgis:5432` whereas on the host it can be found at `localhost:5439`).

So rather than specifying `host`/`port`/`database`/etc, just set `service` to `qwc_geodb` and add a matching service definition to your host `~/.pg_service.conf`

    [qwc_geodb]
    host=localhost
    port=5439
    dbname=qwc_demo
    user=qwc_service
    password=qwc_service
    sslmode=disable

**Comment by jshirokov-bs** [comment]
@manisandro 
So, if i get this correct, i need to do this pipeline:
1. I need to change the connection in the file, according to the configuration from your message
2. In the QGIS-PostGIS connection, specify the service - qwc_geodb

But after that, unfortunately, I can't connect to the database.
I think I'm foolishly missing something.
![image](https://user-images.githubusercontent.com/117968115/228260154-fa87901b-1d2a-4454-ba96-293ea929d6a8.png)

**Comment by manisandro** [comment]
QGIS isn't picking up your service definition. Either you have not restarted QGIS since changing your pg_service.conf, or you have added your service definition to file which is not picked up by the postgres library. The default location for the pg service configuration file for your user is `~/.pg_service.conf` aka `/home/$USER/.pg_service.conf`.

(If you specify service, you don't need to specify host/port/database/user/password as they are specified in the service definition.)

**Comment by jshirokov-bs** [comment]
@manisandro 
Is it a file that is located in the qwc-doker directory, or is it some other file with the .conf extension?
![image](https://user-images.githubusercontent.com/117968115/228269619-a337e33d-69dc-48a3-be8a-c797b25a841d.png)

**Comment by jshirokov-bs** [comment]
@manisandro 
But I don 't ~/.pg_service.conf in the directory /home/$USER/. Should I add it manually?
![image](https://user-images.githubusercontent.com/117968115/228434428-bae3bf04-311d-4caa-80ed-8541183e67a6.png)

**Comment by manisandro** [comment]
Yes

**Comment by jshirokov-bs** [comment]
@manisandro 
okey, so
i create /.pg_service.conf in /home/$USER/. I added this text:
```
[qwc_geodb]
host=localhost
port=5439
dbname=qwc_demo
user=qwc_service
password=qwc_service
sslmode=disable
```
After that, I restarted QWC2 and tried to connect through the "Service". Nothing has changed.
![image](https://user-images.githubusercontent.com/117968115/228500641-71da54a3-25e7-410d-9b6b-c1af2c79c927.png)

**Comment by tpo** [comment]
In order to test your connection to the local DB start with psql (and not QGIS).

So from the commandline do this:

    PGSERVICE=qwc_geodb psql

Only after you are able to connect in this way via psql to your local DB try to connect to it from QGIS.

**Comment by jshirokov-bs** [comment]
@tpo 
I got this result. This doesnt look helpful.
![image](https://user-images.githubusercontent.com/117968115/228524987-2a45132f-1984-4cb5-856e-a0c085c2120c.png)

**Comment by tpo** [comment]
It is helpful :-D - it tells you that your DB and PGSERVICE configuration indeed works!

**Comment by tpo** [comment]
I am guessing from the screenshots you posted: you are running QGIS on a windows machine? If that is the case then you need to put the pg_service.conf file under `C:\Users\<YOUR_USER_NAME_HERE>\AppData\postgresql\.pg_service.conf`

**Comment by tpo** [comment]
Also what is hidden by the "Enter Credentials" window? There are authentication settings that are not visible in the screenshot

**Comment by jshirokov-bs** [comment]
@tpo 
It worked! Now I can connect via the service.
The question remains with editable layers. 
I need to copy a layer from a demo project and place .qgs in "scan", then reconfigure the service, click "import maps" in resources... and then? 
Or am I incorrectly describing the sequence of actions?
![image](https://user-images.githubusercontent.com/117968115/228540257-ab5a4349-60d4-4eb8-8ab9-eb34a73a75d3.png)

**Comment by jshirokov-bs** [comment]
When I try to add an editable layer from demo to the natural-earth-countries project, I get the following message in the log:
![image](https://user-images.githubusercontent.com/117968115/228558392-9b0f2780-7e72-411d-b88e-558742d86361.png)

**Comment by RodolfoMontenegro** [comment]
Hello, I'm having the same issue, while trying to load project and its layers from the postgis connections.
I manually configures the themesConfig.json to add the project.
And when i run the config generator I get the following error.

![error](https://github.com/qwc-services/qwc-docker/assets/139581851/cc7666c1-f67d-41b1-b790-94bb18cb4798)

**Comment by manisandro** [comment]
That error message is most likely accurate. Verify that the credentials you have specified for `qgisprojects` in you `pg_service.conf` do indeed have read access to `qwc_geodb.qgis_projects`.

**Comment by RodolfoMontenegro** [comment]
@manisandro 
Thank you for your quick response!

I'm following the guide for adding projects from:
https://qwc-services.github.io/master/configuration/ThemesConfiguration/#ensuring-valid-datasources

https://qwc-services.github.io/master/configuration/ThemesConfiguration/#storing-projects-in-database

Where it says : Add a qgisprojects service definition for the used database to qwc-docker/pg_service.conf.

So this are my services:
![pg_service](https://github.com/qwc-services/qwc-docker/assets/139581851/c0a76076-f236-4881-afd2-be377f16ede7)

**Comment by HusseinKabbout** [comment]
@jshirokov-bs Were you able to fix your issue?

@RodolfoMontenegro Please open a new issue for your problem.

**Comment by RodolfoMontenegro** [comment]
> @RodolfoMontenegro Please open a new issue for your problem.

Thank you @HusseinKabbout , I was able to fix the issue.
